### ISSUE TITLE
Why doesn't the return value type of `get` include `undefined`?

### ISSUE THREAD

**Issue by evrimfeyyaz** [jakearchibald/idb-keyval#82]
Is there a reason why `get` is defined like this?

```ts
export function get<Type>(key: IDBValidKey, store = getDefaultStore()): Promise<Type>
```

&nbsp;
If the key doesn't exist in IDB, the return value is `undefined`. I personally think:

```ts
export function get<Type>(key: IDBValidKey, store = getDefaultStore()): Promise<Type | undefined>
```

is a better way to define it. This way, you can check if the return value is `undefined` to see if the get operation succeeded.

Am I missing something?

**Comment by jakearchibald** [comment]
You're right.

Fwiw, I think the pattern of passing in the type like this is pretty useless. In the next version I'd like to switch to a model more like [IDB](https://github.com/jakearchibald/idb#typescript), where a single interface defines all the stored types, then the key name can autocomplete too.

Passing in the type just to get it out the other side is pretty pointless.

**Comment by evrimfeyyaz** [comment]
@jakearchibald That's definitely a valid way to go, although from my biased and limited perspective, I'd say keeping the library as similar to Local Storage as possible has its merit too.

What caught my attention in this library was its simplicity and size. Pretty much all I had to do was replace my `localStorage.setItem` and `localStorage.getItem` calls.

**Comment by jakearchibald** [comment]
Ah, I'm only talking about changing how typing works, the API itself will remain as simple as it is.

**Comment by evrimfeyyaz** [comment]
I see, got it! Love both idb-keyval and idb, thanks for the great work.

**Comment by jakearchibald** [comment]
Fixed in 60294946c4e634fcfba340667a55e39500945f7b
